common.sh

```diff
@@ -49,8 +49,8 @@ NODES_PLATFORM=${NODES_PLATFORM:-"libvirt"}
 MASTER_NODES_FILE=${MASTER_NODES_FILE:-"ocp/master_nodes.json"}
 
 export RHCOS_IMAGE_URL=${RHCOS_IMAGE_URL:-"https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa/"}
-export RHCOS_LATEST="$(curl ${RHCOS_IMAGE_URL}/builds.json | jq -r ".builds[0]")"
-export RHCOS_IMAGE_VERSION="${RHCOS_IMAGE_VERSION:-${RHCOS_LATEST}}"
+export RHCOS_INSTALLER_PIN=$(jq -r '.buildid' $GOPATH/src/github.com/openshift-metalkube/kni-installer/data/data/rhcos.json)
+export RHCOS_IMAGE_VERSION="${RHCOS_IMAGE_VERSION:-${RHCOS_INSTALLER_PIN}}"
```

Comment:
We source common.sh before we sync the repo in `03`, will `set -e` cause us to exit because of it?

Reply:
Good question - I only worried about the first use OF RHCOS_IMAGE* which is in 04

Reply:
Maybe we could move this to a separate include, and only where we use it? It calls `curl` every time it's sourced, even when we don't need any of the RHCOS_* vars.

Reply:
Yeah, this likely needs a bunch of hacking to make work properly ... I'll just leave this here as is for now for feedback on the general idea

Reply:
Stephen - if you want to take this over, feel free

Reply:
This currently won't work because we're missing a `eval "$(go env)"` in common.sh, otherwise looks like a good approach to me

Reply:
I'm happy to use this approach as well, my point was on a brand  new fresh environment without a checked out installer it's going to fail, because the file doesn't exist and common.sh gets sourced before anything else.  For most of us and on CI, there's already a checked out installer so it's not going to be obvious until later.

Reply:
I'd support moving it out of common.sh. The image server reliably slows down to the point where it's often unreachable every Friday afternoon, and it's a pain that that blocks stuff that doesn't even need to know about images.

FWIW this doesn't seem to break when the path is not found though (accidentally tested because with no GOPATH set, the file cannot be found). It kept on running up to the point where it tried to run commands on the image.

Reply:
Oh, weird.  `set -xe` doesn't fail with export.

```
#!/bin/bash
set -xe

# doesn't fail
export HELLO=$(jq notexist.json)

# fail
WORLD=$(jq notexist.json)
```

Ignore me then :-) I think separating images could be separate, let's add go env to this